codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -139,4 +147,20 @@ private void constructHandler() {
                 "NewClientHandler", GoDependency.SMITHY_HTTP_TRANSPORT).build();
         writer.write("handler := $T($T(options.HTTPClient), stack)", decorateHandler, newClientHandler);
     }
+
+
+    /**
+     * populateOperationMiddlewareStack adds middleware to the operation middleware stack.
+     */
+    private void populateOperationMiddlewareStack() {
+        writer.write("");
+        String serializerMiddlewareName = ProtocolGenerator.getSerializeMiddlewareName(operation.getId(),
+                protocolGenerator.getProtocolName());
+        writer.write("stack.Serialize.Add($L{}, middleware.After)", serializerMiddlewareName);
```

Comment:
If the methods on the middleware are pointers probably need to take a `&` of the struct being initialized for the serialize/deserialize.